### PR TITLE
[Feat] Added support for Laravel attribute setters in Laravel preset

### PR DESCRIPTION
**Pull request by olivernybroe**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | yes
| Fixed tickets | #135 

This PR adds support for the Laravel set attribute syntax without showing it as an error.

This has been implemented by creating a custom Sniff for setters which replaces the one we used from a package. This new sniff has support for filtering methods based on a regex.

In the Laravel Preset the regex for Laravel attribute setters has been added in as methods which to ignore.

resolves: #135

### CODE REVIEW

**Comment by olivernybroe** [comment]
@nunomaduro This PR also adds an example for testing that the Laravel Preset actually ignores attribute setters. It is not a bullet proof way, as it just checks if any errors are present in the `Classes` metric.

**Comment by nunomaduro** [comment]
Sounds awesome, going to check this tonight.

File: phpunit.xml.dist [hunk 1]
```diff
@@ -3,7 +3,7 @@
          backupStaticAttributes="false"
          beStrictAboutTestsThatDoNotTestAnything="true"
          beStrictAboutOutputDuringTests="true"
-         bootstrap="vendor/autoload.php"
+         bootstrap="vendor/squizlabs/php_codesniffer/tests/bootstrap.php"
```
[the remark this answers is not in the document]
Comment (olivernybroe) [reply phpunit.xml.dist]: Yes, in Squizlabs autoloader it also loads composer autoloader if present. https://github.com/squizlabs/PHP_CodeSniffer/blob/master/autoload.php#L66-L68

**Comment by nunomaduro** [comment]
@olivernybroe Thanks for this pull request. Actually I think we can do better on this: A generic feature where on the configuration you can ignore errors globally or on specific classes. What do you think?

**Comment by olivernybroe** [comment]
@nunomaduro I like the idea of adding that in to ignore errors in a more broad way. However for it to work good enough to replace this sniff, it has to be specific enough to I can ignore an error based on the naming of a method with a regex.

This PR still disallows setters in models, but just ignores them if they are laravel setters. The Sniff itself just has support for ignoring methods based on a regex.

My opinion would be to first merge this in, as we are going to need custom sniff in some cases and fixing an issue like this is pretty important as a lot of users are using laravel mutators.
Then we can work on setting up a generic way to ignore classes.

**Comment by nunomaduro** [comment]
@olivernybroe Alright! Can you just check why tests are not passing?

**Comment by nunomaduro** [comment]
Very good job! Thanks for this @olivernybroe. Fell free to perform the next release: https://github.com/nunomaduro/phpinsights/blob/master/RELEASE.md. Please ask me if you have any questions.

**Pull request merged**